src/internal/m365/collection/drive/item.go

```diff
@@ -33,39 +35,39 @@ var downloadURLKeys = []string{
 
 func downloadItem(
 	ctx context.Context,
-	ag api.Getter,
+	getter api.Getter,
+	driveID string,
 	item *custom.DriveItem,
 ) (io.ReadCloser, error) {
 	if item == nil {
 		return nil, clues.New("nil item")
 	}
 
 	var (
-		rc     io.ReadCloser
-		isFile = item.GetFile() != nil
+		// very large file content needs to be downloaded through a different endpoint, or else
+		// the download could take longer than the lifespan of the download token in the cached
+		// url, which will cause us to timeout on every download request, even if we refresh the
+		// download url right before the query.
+		url    = "https://graph.microsoft.com/v1.0/" + driveID + "/items/" + ptr.Val(item.GetId()) + "/content"
+		reader io.ReadCloser
 		err    error
 	)
 
-	if isFile {
-		var (
-			url string
-			ad  = item.GetAdditionalData()
-		)
-
-		for _, key := range downloadURLKeys {
-			if v, err := str.AnyValueToString(key, ad); err == nil {
-				url = v
-				break
-			}
-		}
+	// if this isn't a file, no content is available for download
+	if item.GetFile() == nil {
+		return reader, nil
+	}
 
-		rc, err = downloadFile(ctx, ag, url)
-		if err != nil {
-			return nil, clues.Stack(err)
-		}
+	// smaller files will maintain our current behavior (prefetching the download url with the
+	// url cache).  That pattern works for us in general, and we only need to deviate for very
+	// large file sizes.
+	if ptr.Val(item.GetSize()) > largeFileDownloadLimit {
```

Comment:
Should this be < `largeFileDownloadLimit`?

Reply:
whoops, yes, mixed that up from undoing manual testing.